src/ToDo/Business/ToDoTask.cs

```diff
@@ -15,6 +15,11 @@ internal ToDoTask(TaskList list, TaskData data)
 	{
 	}
 
+	internal ToDoTask(TaskData data)
+	: this(data?.ParentList?.Id!, data!)
```

Comment:
Double `?` ending with a `!` sounds really really weird. Either we know that the `data`, the `ParentList` and the `Id` won't be `null`, either we have to support that the `listId` can be null ...

Reply:
mm you're right, but parentList can be null. So, I removed one `?`

Reply:
Well if the `ParentList` can be null, it means that you should just not use the `!` at the end and you have to allow the `listId` parameter to be `null` or just explicitly throw an exception (`data?.ParentList?.Id ?? throw new ArgumentException("The taskData does not have a listId")`. It's not a matter of number of `?` it's the fact you are mixing them with `!`, either it can be null, either it can't.

BTW I would imagine that the API **always** provides the `ParentList` and this will always have an `Id` so can probably flag those as non nullable???

Reply:
No, the API only provides the PArentList in two endpoints. The new endpoints. In other case the ParentList is null.

Reply:
throw an exception in the constructor? 

I think that `!` operator is only for null forgiving. So, if `data.ParentList?.Id!` it's good, because I only use `?` for verify de nullability of ParentList  and `!` for null forgiving of `ParentList.Id`. 

In other case do you prefer that throw the exception?

Reply:
Yes an exception in the ctor is not really a good pattern, but definitively better than a null ref when using a non nullable value. Nullable annotations are here to help us to detect and handle null values, but here you are just dismissing it and hiding a potential issue that will be much more complicated to debug as we would not expect a `null` from the non nullable `ToDoTask.ListId`. So yes I personally prefer to throw an exception at the earliest location where we can detect it, and throw meaning full exceptions instead of an obscure `null ref`.

My preference would be to throw an explicit exception as mentioned `data?.ParentList?.Id ?? throw new ArgumentException("The TaskData does not have a listId. This ctor should be used only by 'new endpoints'.")`. But it could also be a raw "null ref" by using `data.ParentList!.Id!` : not much more elegant, but as least it will fail at the location where we wrongly use it.

And personally to avoid throwing in the ctor, I would just remove that ctor which assumes that it's being invoked only by some specific users (i.e. "new endpoints"). Then I would do the `null` check in the caller: 
```csharp
var listId = data.ParentList?.Id ?? throw new InvalidOperationException("The API did not provide list information.");
return new ToDoTask(listId, data);
```

Reply:
Okay. Thank you

Reply:
Changed it

Reply:
The change implemented doesn't make sense - there's no requirement that a task has to belong to a list, so the ParentList value can be null